consensus/hotstuff/safetyrules/safety_rules.go

```diff
@@ -129,6 +129,12 @@ func (r *SafetyRules) ProduceVote(proposal *model.Proposal, curView uint64) (*mo
 func (r *SafetyRules) ProduceTimeout(curView uint64, newestQC *flow.QuorumCertificate, lastViewTC *flow.TimeoutCertificate) (*model.TimeoutObject, error) {
 	lastTimeout := r.safetyData.LastTimeout
 	if lastTimeout != nil && lastTimeout.View == curView {
+		// increment timeout tick, since we are using a pointer this change will be reflected in `safetyData` as well.
+		lastTimeout.TimeoutTick++
+		err := r.persist.PutSafetyData(r.safetyData)
+		if err != nil {
+			return nil, fmt.Errorf("could not persist safety data: %w", err)
+		}
 		return lastTimeout, nil
```

Comment:
## ❓❗❓ 
I am worried about the lack of concurrency control here. We are passing the `TimeoutObject` as a pointer through the consensus node. I don't think we should be modifying these structs for consistency reasons. I would suggest to treat `TimeoutObject` as immutable structs and make a copy here:

```suggestion
		// model.TimeoutObject are conceptually immutable, hence we create a shallow copy here, which allows us to increment TimeoutTick
		var updatedTimeout model.TimeoutObject = *lastTimeout
		updatedTimeout.TimeoutTick += 1

		// persist updated TimeoutObject in `safetyData` and return it
		r.safetyData.LastTimeout = &updatedTimeout
		err := r.persist.PutSafetyData(r.safetyData)
		if err != nil {
			return nil, fmt.Errorf("could not persist safety data: %w", err)
		}
		return &updatedTimeout, nil
```